Sources/Core/AWSClientRuntime/EventStream/AWSMessageEncoder.swift

```diff
@@ -0,0 +1,23 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import ClientRuntime
+
+extension AWSEventStream {
+
+    /// Encodes a `Message` into a `Data` object
+    /// to be sent over the wire.
+    public struct AWSMessageEncoder: MessageEncoder {
+        init() {}
+
+        /// Encodes a `Message` into a `Data` object
+        public func encode(message: EventStream.Message) throws -> Data {
+            let crtMessage = message.toCRTMessage()
+            return try crtMessage.getEncoded()
```

Comment:
Ideally this should be injected in to remove the tight coupling to CRT. This would allow us to easily use different logic in the future for encoding (not CRT) and it would allow us to easily mock this if we wanted to. 

I'm also fine with this being out of scope, but still wanted to mention.

Reply:
we always swap the implementation without changing the current interfaces. This is implementation details here.